src/main/java/com/blazemeter/ciworkflow/CiPostProcess.java

```diff
@@ -122,15 +133,85 @@ public boolean errorsFailed(JSONArray errors) {
     /**
      * Saves junit report to hdd;
      */
-    public void saveJunit() {
+    public void saveJunit(Master master) {
+        try {
+            String junitReport = master.getJUnitReport();
+            File junitFile = createJunitFile(junitPath + File.separator + master.getId() + ".xml",
+                    workspaceDir);
+            Files.write(Paths.get(junitFile.toURI()), junitReport.getBytes());
+        } catch (Exception e) {
+            notifier.notifyAbout("Failed to save junit report from master = " + master.getId() + " to disk.");
+        }
+    }
 
+    private File createJunitFile(String junitPath, String workspaceDir) throws Exception {
+        File junitFile = new File(junitPath);
+        try {
+            junitFile.createNewFile();
+        } catch (IOException e) {
+            junitFile = new File(workspaceDir);
+            junitFile.createNewFile();
+        }
+        return junitFile;
     }
 
     /**
      * Saves jtl report to hdd;
      */
-    public void saveJtl() {
+    public void saveJtl(Master master) {
+        URL url = null;
+        try {
+            List<Session> sessions = master.getSessions();
+            for (Session s : sessions) {
+                url = new URL(s.getJTLReport());
+                downloadUnzip(url);
+            }
+        } catch (Exception e) {
+            notifier.notifyAbout("Unable to get JTLZIP from " + url + " " + e);
+        }
+    }
 
+    public void downloadUnzip(URL url) {
+        int i = 1;
```

Comment:
Does `i` means `attempNumber`? 
Can we use `for` cycle with `break;` for exit from the loop?